ethpm_cli/_utils/solc.py

```diff
@@ -43,12 +47,53 @@ def generate_solc_input(contracts_dir: Path) -> None:
         contracts_dir.parent,
         SOLC_INPUT,
     )
-    cli_logger.info(
-        f"Use `solc --standard-json --allow-paths /[ABS_PATH_TO]/{contracts_dir} "
-        f"< /[REL_PATH_TO]/{SOLC_INPUT} > /[REL_PATH_TO]/{contracts_dir.parent / SOLC_OUTPUT}` "
-        "to generate the Solidity compiler output. "
-        "Requires that you have the correct Solidity compiler version installed."
+
+
+def compile_contracts(project_dir: Path) -> None:
+    contracts_dir = project_dir / "contracts"
+    contracts = [contract.name for contract in contracts_dir.glob("**/*.sol")]
+    if len(contracts) == 0:
+        raise CompilationError(f"No Solidity contracts found in {contracts_dir}.")
+    contracts_display = "\n- ".join(contracts)
+    cli_logger.info("Compiling contracts found in %s", contracts_dir)
+    cli_logger.info("%d contracts found: \n- %s", len(contracts), contracts_display)
+
+    tmp_project_dir = Path(tempfile.TemporaryDirectory().name) / "project"
```

Comment:
You probably want to be using the context manager interface for the temporary directory so that it automatically gets cleaned up.